README.md

```diff
@@ -2,217 +2,170 @@
+More information can be found [here](https://docs.openshift.com/rosa/welcome/index.html).
 
-provider "ocm" {
-  token = "..."
-}
+## AWS STS
+A Secure Token Service (STS) is a component that issues, validates, renews, and cancels security tokens for trusted systems, users, and resources requesting access within a federation.
+AWS provides AWS STS as a web service that enables you to request temporary, limited-privilege credentials for AWS Identity and Access Management (IAM) users or for users you authenticate (federated users).
 
-resource "ocm_cluster" "my_cluster" {
-  name           = "my-cluster"
-  cloud_provider = "aws"
-  product        = "osd"
-  cloud_region   = "us-east-1"
-}
+## ROSA STS mode
 
-resource "ocm_identity_provider" "my_idp" {
-  cluster = ocm_cluster.my_cluster.id
-  name    = "my-idp"
-  htpasswd = {
-    username = "admin"
-    password = "redhat123"
-  }
-}
+To deploy a Red Hat OpenShift Service on AWS (ROSA) cluster that uses the AWS Security Token Service (STS), you must create the following AWS Identity Access Management (IAM) resources:
 
-resource "ocm_group_membership" "my_admin" {
-  cluster = ocm_cluster.my_cluster.id
-  group   = "dedicated-admins"
-  user    = "admin"
-}
-```
+Specific account-wide IAM roles and policies that provide the STS permissions required for ROSA support, installation, control plane, and compute functionality. This includes account-wide Operator policies.
+Cluster-specific Operator IAM roles that permit the ROSA cluster Operators to carry out core OpenShift functionality.
+An OpenID Connect (OIDC) provider that the cluster Operators use to authenticate.
+
+## Terraform
 
-The value of the `token` attribute of the provider should be the OCM
-authentication token that you can get [here](https://console.redhat.com/openshift/token).
-If this attribute isn't used then the provider will try to get the token it from
-the `OCM_TOKEN` environment variable.
+Terraform is an infrastructure as a code tool, used primarily by DevOps teams.
+It lets you define resources in human-readable configuration files that you can version, reuse, and share.
 
-### Create AWS Rosa STS Cluster
+Terraform creates and manages resources through their application programming interfaces (APIs) by using "Providers".
 
-The following example shows a production grade rosa cluster with:
+## Prerequisites
 
-* Existing VPC & Subnets
-* Multi AZ
-* Proxy
-* STS
+In order to use the provider inside your terraform configuration you need to import it using:
+
+* Offline token (OCM):
+
+Get an offline token from [https://console.redhat.com/openshift/token/rosa](https://console.redhat.com/openshift/token/rosa)
+
+* Create ROSA account IAM roles: 
+
+Detailed ROSA Account Roles and Policies can be found [here](https://docs.openshift.com/rosa/rosa_architecture/rosa-sts-about-iam-resources.html)
 
 ```
-data "aws_caller_identity" "current" {}
+rosa create account-roles
```

Comment:
I think better to rely on aws cli commands than rosa cli 
https://docs.google.com/document/d/1HTbg8Z_7PJXDtJ-Funm-NWaZZpRC6BJvKZk0hiAbfAU/edit#heading=h.p89ii1gka5cs